# Simulacion
Libreria random

In [3]:
#librerias
import random
import collections
from tabulate import tabulate


In [17]:
total=[]
lista=[]
print('')
eventos=100
print(eventos,' Eventos')
print('','\n')

for j in range(eventos):
    numero1=random.randint(1, 6)
    lista.append(numero1)
    numero2=random.randint(1, 6)
    lista.append(numero2)
    suma=numero1+numero2
    total.append(suma)
    
cuenta1 = collections.Counter(total)    


lista1 = list(set(cuenta1.elements()))
lista1.sort()
lista2=[]


print('Suma','Frecuencia','Probabilidad')
for i in range(len(lista1)):
    a=lista1[i]
    lista2.insert(0,a)
    lista2.insert(1,cuenta1[a])
    lista2.insert(2,cuenta1[a]/eventos)
    print('',a,'      ',cuenta1[a],'     ',cuenta1[a]/eventos)
    


100  Eventos
 

Suma Frecuencia Probabilidad
 2        4       0.04
 3        4       0.04
 4        9       0.09
 5        6       0.06
 6        17       0.17
 7        15       0.15
 8        16       0.16
 9        12       0.12
 10        10       0.1
 11        6       0.06
 12        1       0.01


In [16]:
total=[]
lista=[]
print('')
eventos=1000
print(eventos,' Eventos')
print('','\n')

for j in range(eventos):
    numero1=random.randint(1, 6)
    lista.append(numero1)
    numero2=random.randint(1, 6)
    lista.append(numero2)
    suma=numero1+numero2
    total.append(suma)
    
cuenta1 = collections.Counter(total)    


lista1 = list(set(cuenta1.elements()))
lista1.sort()
lista2=[]


print('Suma','Frecuencia','Probabilidad')
for i in range(len(lista1)):
    a=lista1[i]
    lista2.insert(0,a)
    lista2.insert(1,cuenta1[a])
    lista2.insert(2,cuenta1[a]/eventos)
    print('',a,'      ',cuenta1[a],'     ',cuenta1[a]/eventos)
    


1000  Eventos
 

Suma Frecuencia Probabilidad
 2        26       0.026
 3        55       0.055
 4        83       0.083
 5        92       0.092
 6        151       0.151
 7        169       0.169
 8        137       0.137
 9        108       0.108
 10        86       0.086
 11        71       0.071
 12        22       0.022


In [15]:
total=[]
lista=[]
print('')
eventos=10000
print(eventos,' Eventos')
print('','\n')

for j in range(eventos):
    numero1=random.randint(1, 6)
    lista.append(numero1)
    numero2=random.randint(1, 6)
    lista.append(numero2)
    suma=numero1+numero2
    total.append(suma)
    
cuenta1 = collections.Counter(total)    


lista1 = list(set(cuenta1.elements()))
lista1.sort()
lista2=[]


print('Suma','Frecuencia','Probabilidad')
for i in range(len(lista1)):
    a=lista1[i]
    lista2.insert(0,a)
    lista2.insert(1,cuenta1[a])
    lista2.insert(2,cuenta1[a]/eventos)
    print('',a,'      ',cuenta1[a],'     ',cuenta1[a]/eventos)
    


10000  Eventos
 

Suma Frecuencia Probabilidad
 2        286       0.0286
 3        552       0.0552
 4        815       0.0815
 5        1125       0.1125
 6        1373       0.1373
 7        1728       0.1728
 8        1320       0.132
 9        1107       0.1107
 10        805       0.0805
 11        600       0.06
 12        289       0.0289



# Simulacion
Libreria numpy



In [50]:
import numpy as np


In [54]:
total=[]
lista=[]
print('')
eventos=100
print(eventos,' Eventos')
print('','\n')
for j in range(eventos):
    numero1=np.random.randint(1,7)
    numero2=np.random.randint(1,7)
    lista.append(numero1)
    lista.append(numero2)
    suma=numero1+numero2
    total.append(suma)
    
cuenta1 = collections.Counter(total)    


lista1 = list(set(cuenta1.elements()))
lista1.sort()
lista2=[]


print('Suma','Frecuencia','Probabilidad')
for i in range(len(lista1)):
    a=lista1[i]
    lista2.insert(0,a)
    lista2.insert(1,cuenta1[a])
    lista2.insert(2,cuenta1[a]/eventos)
    print('',a,'      ',cuenta1[a],'     ',cuenta1[a]/eventos)


100  Eventos
 

Suma Frecuencia Probabilidad
 2        3       0.03
 3        3       0.03
 4        8       0.08
 5        19       0.19
 6        12       0.12
 7        13       0.13
 8        13       0.13
 9        12       0.12
 10        8       0.08
 11        6       0.06
 12        3       0.03


In [55]:
total=[]
lista=[]
print('')
eventos=1000
print(eventos,' Eventos')
print('','\n')
for j in range(eventos):
    numero1=np.random.randint(1,7)
    numero2=np.random.randint(1,7)
    lista.append(numero1)
    lista.append(numero2)
    suma=numero1+numero2
    total.append(suma)
    
cuenta1 = collections.Counter(total)    


lista1 = list(set(cuenta1.elements()))
lista1.sort()
lista2=[]


print('Suma','Frecuencia','Probabilidad')
for i in range(len(lista1)):
    a=lista1[i]
    lista2.insert(0,a)
    lista2.insert(1,cuenta1[a])
    lista2.insert(2,cuenta1[a]/eventos)
    print('',a,'      ',cuenta1[a],'     ',cuenta1[a]/eventos)


1000  Eventos
 

Suma Frecuencia Probabilidad
 2        33       0.033
 3        57       0.057
 4        81       0.081
 5        112       0.112
 6        142       0.142
 7        170       0.17
 8        140       0.14
 9        100       0.1
 10        76       0.076
 11        57       0.057
 12        32       0.032


In [56]:
total=[]
lista=[]
print('')
eventos=10000
print(eventos,' Eventos')
print('','\n')
for j in range(eventos):
    numero1=np.random.randint(1,7)
    numero2=np.random.randint(1,7)
    lista.append(numero1)
    lista.append(numero2)
    suma=numero1+numero2
    total.append(suma)
    
cuenta1 = collections.Counter(total)    


lista1 = list(set(cuenta1.elements()))
lista1.sort()
lista2=[]


print('Suma','Frecuencia','Probabilidad')
for i in range(len(lista1)):
    a=lista1[i]
    lista2.insert(0,a)
    lista2.insert(1,cuenta1[a])
    lista2.insert(2,cuenta1[a]/eventos)
    print('',a,'      ',cuenta1[a],'     ',cuenta1[a]/eventos)


10000  Eventos
 

Suma Frecuencia Probabilidad
 2        284       0.0284
 3        585       0.0585
 4        818       0.0818
 5        1070       0.107
 6        1370       0.137
 7        1645       0.1645
 8        1431       0.1431
 9        1124       0.1124
 10        839       0.0839
 11        561       0.0561
 12        273       0.0273


# Conclusión

Se desarrollo un modelo para la simulación para la interrogante de:
¿Cuál es la probabilidad de lanzar dos dados y que la suma de sus caras superiores sea 2,3,4,5,6,7,8,9,10,11,12?
Si fuera el caso de un juego de apuestas, ¿usted ah que numero apostaría?

Con 100 lanzamientos:
Suma=6        repeticiones=17       probabilidad=0.17

Con 1000 lanzamientos:
Suma=7        repeticiones=169       probabilidad=0.169

Con 10000 lanzamientos:
Suma=7        repeticiones=1728       probabilidad=0.1728

Una vez realizada la simulación podemos concluir que los números más probables en salir son los números intermedios, en este caso la probabilidad más grande es de la suma de 6 y 7
